python/ray/dashboard/agent.py

```diff
@@ -180,38 +180,43 @@ async def run(self):
 
         modules = self._load_modules()
 
+        launch_http_server = True
         if self.http_server:
             try:
                 await self.http_server.start(modules)
-            except Exception:
-                # TODO(SongGuyang): Catch the exception here because there is
-                # port conflict issue which brought from static port. We should
-                # remove this after we find better port resolution.
+            except Exception as e:
+                # TODO(kevin85421): We should fail the agent if the HTTP server
+                # fails to start to avoid hiding the root cause. However,
+                # agent processes are not cleaned up correctly after some tests
+                # finish. If we fail the agent, the CI will always fail until
+                # we fix the leak.
```

Comment:
We should definitely follow up and make this change, perhaps @MortalHappiness can take it?

Reply:
@MortalHappiness would you mind opening an issue to track the progress? After you fix the issue, https://github.com/ray-project/ray/pull/51705 can be merged.

Reply:
Created issue: https://github.com/ray-project/ray/issues/51925

@kevin85421 Do you mean that I directly take over #51705 and fix the test error? And then you'll close #51705?

Reply:
No, there are two issues.

1. dashboard agent leaks --> you take this one
2. fail dashboard agent after (1) fixes --> i.e. #51705.